distributed/scheduler.py

```diff
@@ -1450,7 +1454,10 @@ def who_wants(self):
 
     @property
     def who_has(self):
-        return self._who_has
+        try:
```

Comment:
As a small short term hack I added this try/except because we will sometime return WorkerState objects in the dictionary:

```
ipdb> WhoHas(self._who_has)
*** TypeError: cannot convert dictionary update sequence element #0 to a sequence
ipdb> pp self._who_has
{<WorkerState 'tcp://127.0.0.1:57682', name: 0, memory: 1, processing: 0>}
ipdb>
```

You can see this in the `distributed/tests/test_client.py::test_idempotence` test